src/OSPSuite.Core/Domain/Mappers/EventAssignmentBuilderToEventAssignmentMapper.cs

```diff
@@ -1,37 +1,61 @@
+using System.Collections.Generic;
+using System.Linq;
 using OSPSuite.Core.Domain.Builder;
 
 namespace OSPSuite.Core.Domain.Mappers
 {
    /// <summary>
-   /// Maps one event assignment from building block into model
+   ///    Maps one event assignment from building block into model
    /// </summary>
-   public interface IEventAssignmentBuilderToEventAssignmentMapper : IBuilderMapper<IEventAssignmentBuilder, IEventAssignment>
+   public interface IEventAssignmentBuilderToEventAssignmentMapper : IBuilderMapper<IEventAssignmentBuilder, IReadOnlyList<IEventAssignment>>
    {
    }
 
-   class EventAssignmentBuilderToEventAssignmentMapper : IEventAssignmentBuilderToEventAssignmentMapper
+   public class EventAssignmentBuilderToEventAssignmentMapper : IEventAssignmentBuilderToEventAssignmentMapper
    {
       private readonly IObjectBaseFactory _objectBaseFactory;
       private readonly IFormulaBuilderToFormulaMapper _formulaMapper;
 
-      public EventAssignmentBuilderToEventAssignmentMapper(IObjectBaseFactory objectBaseFactory,IFormulaBuilderToFormulaMapper formulaMapper)
+      public EventAssignmentBuilderToEventAssignmentMapper(IObjectBaseFactory objectBaseFactory, IFormulaBuilderToFormulaMapper formulaMapper)
       {
          _objectBaseFactory = objectBaseFactory;
          _formulaMapper = formulaMapper;
       }
 
-      public IEventAssignment MapFrom(IEventAssignmentBuilder assignmentBuilder, IBuildConfiguration buildConfiguration)
+      public IReadOnlyList<IEventAssignment> MapFrom(IEventAssignmentBuilder assignmentBuilder, IBuildConfiguration buildConfiguration)
+      {
+         if (!isForAllFloating(assignmentBuilder))
+            return new[] {createAssignment(assignmentBuilder, buildConfiguration)};
+
+         return buildConfiguration.Molecules.AllFloating()
+            .Select(x => createMoleculeAssignment(x, assignmentBuilder, buildConfiguration))
+            .ToList();
+      }
+
+      private IEventAssignment createMoleculeAssignment(IMoleculeBuilder moleculeBuilder, IEventAssignmentBuilder assignmentBuilder, IBuildConfiguration buildConfiguration)
+      {
+         var name = $"{assignmentBuilder.Name}_{moleculeBuilder.Name}";
+         var assignment = createAssignment(assignmentBuilder, buildConfiguration, name);
+         assignment.ObjectPath.Replace(ObjectPathKeywords.ALL_FLOATING, moleculeBuilder.Name);
```

Comment:
Creating one for each omolecule floating. To ensure unicity, I add the name of the molecule at the end of the original name